core/src/test/scala/unit/kafka/log/LogSegmentsTest.scala

```diff
@@ -35,6 +39,17 @@ class LogSegmentsTest {
     LogTestUtils.createSegment(offset, logDir, indexIntervalBytes, time)
   }
 
+  @BeforeEach
+  def setup(): Unit = {
+    logDir = TestUtils.tempDir()
+  }
+
+  @AfterEach
+  def teardown(): Unit = {
+    segmentsBuffer.foreach(_.close())
```

Comment:
The segments are already closed from inside `LogSegments#close()` and therefore can be skipped in the `tearDown()`. https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/log/LogSegments.scala#L78